src/main/java/core/basesyntax/OperationStrategyImpl.java

```diff
@@ -0,0 +1,15 @@
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(
+            Map<FruitTransaction.Operation, OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public OperationHandler getHandler(FruitTransaction.Operation operation) {
+        return operationHandlers.get(operation);
```

Comment:
Consider handling the case where the operation is not found in the map. You could throw an exception with an informative message to avoid returning null which could lead to NullPointerException elsewhere in the code.